src/machine/machined-varlink.c

```diff
@@ -507,6 +508,126 @@ static int vl_method_terminate(sd_varlink *link, sd_json_variant *parameters, sd
         return lookup_machine_and_call_method(link, parameters, flags, userdata, vl_method_terminate_internal);
 }
 
+static int list_image_one(sd_varlink *link, Image *image, bool more) {
+        int r;
+
+        assert(link);
+        assert(image);
+
+        _cleanup_(sd_json_variant_unrefp) sd_json_variant *v = NULL;
+
+        r = sd_json_buildo(
+                        &v,
+                        SD_JSON_BUILD_PAIR_STRING("name", image->name),
+                        JSON_BUILD_PAIR_STRING_NON_EMPTY("path", image->path),
+                        SD_JSON_BUILD_PAIR("type", SD_JSON_BUILD_STRING(image_type_to_string(image->type))),
+                        SD_JSON_BUILD_PAIR("class", SD_JSON_BUILD_STRING(image_class_to_string(image->class))),
```

Comment:
you could use SD_JSON_BUILD_PAIR_STRING() here to shorten the line, but doesn't really matter